File: backend/models/practice_log.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// User represents a user in your application.
type PracticeLog struct {
	gorm.Model
	ActivityID uint      `gorm:"not null"`
	Count      float64   `json:"count" gorm:"not null"`
	Date       time.Time `json:"date" gorm:"type:date;not null"`
	UserID     uint      `gorm:"not null"`
}

File: frontend/src/types/practiceLog.ts
export interface PracticeLog {
    ID: number;
    CreatedAt: string; // Use string for ISO date-time
    UpdatedAt: string; // Use string for ISO date-time
    DeletedAt: string | null; // Use string for ISO date-time or null
    ActivityID: number;
    count: number;
    date: string; // Use string for ISO date
    UserID: number;
}
